packages/@sanity/portable-text-editor/src/editor/hooks/useSyncValue.ts

```diff
@@ -80,15 +80,22 @@ export function useSyncValue(
         withoutSaving(slateEditor, () => {
           withoutPatching(slateEditor, () => {
             Editor.withoutNormalizing(slateEditor, () => {
+              const hadSelection = !!slateEditor.selection
```

Comment:
If the value has become empty, and the user already had a selection on the previous value we don't want to loose the cursor.